fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java

```diff
@@ -3243,158 +3243,183 @@ public void unsetFields() {
 
         // update profile.
         // return true if profile is updated. Otherwise, return false.
-        public synchronized boolean updateProfile(TReportExecStatusParams params) {
+        // Has to use synchronized to ensure there are not concurrent update threads. Or the done
+        // state maybe update wrong and will lose data. see https://github.com/apache/doris/pull/29802/files.
+        public synchronized boolean updatePipelineStatus(TReportExecStatusParams params) {
+            // The fragment or instance is not finished, not need update
+            if (!params.done) {
+                return false;
```

Comment:
This caused a problem where progress reports for `show load` command is ignored until the entire fragment instance is finished.